### ISSUE TITLE
Website Feature Wish List

### ISSUE THREAD

**Issue by kylegordy** [chadev/Chadev-issues#3]
Hey everyone, we're working on throwing together a site for Chadev and would love some feedback and ideas on features for the site. What would you like chadev.com to look like?
## 

Here's some features I would love to see included:
- List of local devs (not sure how best to go about this, any ideas?)
- List of local dev groups/meetings
- Link to/embed [Chadev google calendar](http://goo.gl/2pFn1p) (Perhaps even have a short recap of the events taking place this week on the homepage. What do you all think?)
- Link to/embed IRC chat (We're currently linking to [kiwirc](https://kiwiirc.com/client/irc.freenode.net/?nick=chadev-?#chadev) on the subreddit. Any thoughts?)
- List of dev resources available around town. (I started a list of some in the sidebar of [our subreddit](http://www.reddit.com/r/chadev) earlier. I'm sure I'm missing stuff, if so please let me know :+1:  )
- Mailing list signup? (Not sure if we would want to do a mailing list or not… I thought it might be cool to send out a weekly email listing all the meetups/events happening that week. Any thoughts?)
- Brief about Chadev on the homepage (Not sure we need a whole about page… Any thoughts?)

**Comment by vendion** [comment]
The list looks good to me!  I know @spyc3r I talked about pull the list of
devs via JS/jQuery, assuming the list doesn't get out of hand then it
shouldn't be too bad performance wise.  We maybe able to make use of some
kind of hosted redis service that periodically updates devs apart of the
Chadev org on Github, as well as a list of groups so it would be a matter
of pulling from there and parsing that data.  We can work out such details
in another issue though.

By should recap I assume you mean displaying data like: event title, date &
time, and the description field?  That would be interesting to see, I am
assuming we can scrap this data from the Ical file (I have never messed
with any ical stuff).

Kiwirc is fine for a web client, I have played with it in the past and
works fairly well.  The only other one I know of is qwebirc which FreeNode
using on http://webchat.freenode.net/

+1 for list of dev resources

I wouldn't do it as a mailing list, the IRC and subreddit is already enough
ways to communicate at this point.  As Chadev grows that may change, that
said I am for the idea of a newsletter.  We could run it under a MailChimp
(free as long as you send less than 2,000 emails in a month or so), or a
read-orly Google Group.  Would need to get a few people who are good at
coming up with copy (not me) to write the emails though.

I don't think we would want more than a brief on the homepage, we can
always add a page that has all the information that people would never read
anyways :P

That is just my $0.02

**Comment by acedrew** [comment]
I've got the calendar stuff figured out, I already refactored what @dryan wrote here: https://github.com/dryan/chadev-events once, and actually google's calendar api just hit v3, making it much easier to work with (json returned, vs gdata bs) 

The trick will be figuring out where and how to store the list of devs that we want to use. I use google contacts for that stuff currently, where you would set a group with the contacts and whatever meta-data you wanted. It's essentially a little document database with a handy ui already written. I'm definitely not married to that, as I have to use third party tools to share management responsibility that way. I'm totally not opposed to just having a massive YAML file that either gets compiled into minified js in the build process of the site, or gets loaded using something like this:https://github.com/nodeca/js-yaml/blob/master/README.md 
Either way, adding yourself would be as simple as a PR.

Thoughts?

**Comment by vendion** [comment]
That is good that the calendar stuff figured out, I just pushed a quick fix
to chadev.github.io site embedding the google calendar on the event page.
 The sooner that can be replaced the better, but what is there now is
better than nothing.

As for the datastore I was thinking we could use something like [Redis
Clould on Heroku](https://addons.heroku.com/rediscloud), and have a script
that runs via a cron job somewhere that pulls a list of devs from here and
update the datastore, this method could also apply for dev groups if we
want to pull them dynamically.  Google Contacts should work for just the
dev information as well.

Honestly didn't think of a static file to begin with, but if we go that
route YAML or Markdown (is there a good library for Markdown in JS?) would
be a preferred syntax, both are fairly small and should be common enough
that if you can't figure out how to add your name/group to the list you
probably shouldn't be on it to begin with.

**Comment by kylegordy** [comment]
@vendion @acedrew Thanks guys for helping out on this project! I'm super excited to keep hacking away at this with you. I'll try to get some more work in tonight on the layout and such. 

Also, for now could we just replicate the dev list layout from [chattdevs.com](chattdevs.com) with just pure markdown and css? At the very least this would allow me to start playing around with how we might mark all this up. Any thoughts?

**Comment by vendion** [comment]
@kylegordy No worries.  Feel free to extend the `default-slim` layout I added last night, having the feature div on pages other than the home page just felt odd and took up a lot of space.  Speaking of the dev list, do you have a favorite grid system that we could use for that?

**Comment by kylegordy** [comment]
@vendion Sounds good, will do. As for grids I'm using [Bourbon Neat](http://neat.bourbon.io) on the site right now. It keeps our markup super clean and is really flexible.

**Comment by dryan** [comment]
What's the use case for markdown in JavaScript?

Why not put the devs into a _data file and output from that?

**Comment by acedrew** [comment]
ahhhhhhhh discussion on profiles is moved here: https://github.com/chadev/Chadev-issues/issues/5 
Let's keep this for more general discussion of the rest of the items on the list. 
@dryan see that thread for this exact conclusion :)

**Comment by vendion** [comment]
Speaking of, shouldn't these have been opened on the chadev.github.io repo instead of here?

**Comment by kylegordy** [comment]
@vendion I thought that listing all our issues here in the issues repo was the point of it :wink:

**Comment by vendion** [comment]
Yeah maybe, and it is fine to have it here, I just figured issues and
discussions about the site would take place on that repo, and branding
related would be there, this would be a general or catch all issue tracker.
 If I'm the minority then, forget I said anything :)

**Comment by kylegordy** [comment]
@vendion I'm just trolling at bit I guess :wink:  We can continue these discussions on the site's repo. To help with confusion as to where to place issues should we delete this repo? Just brainstorming here.

**Comment by vendion** [comment]
I think this repo still has value though, like for community structure
planning, someone pointing out a developer group/event that we are unaware
of, etc.  We should just update the README.md to make things clearer about
where things go.

**Comment by kylegordy** [comment]
@vendion That's a really good point, sounds good to me :thumbsup:

**Comment by vendion** [comment]
Are any of these still open, or is this issue good to close?
